homeassistant/components/hue/v2/binary_sensor.py

```diff
@@ -82,9 +82,9 @@ class HueMotionSensor(HueBinarySensorBase):
     _attr_device_class = BinarySensorDeviceClass.MOTION
 
     @property
-    def is_on(self) -> bool | None:
+    def is_on(self) -> bool:
         """Return true if the binary sensor is on."""
-        return self.resource.motion.motion
+        return self.resource.motion.motion is True
```

Comment:
So this is a bit of a weird one to review right now.

In the current stable release (`2021.12`), it should be a boolean that is returned.
However, on the current `dev` (`2022.2`), binary sensors actually can return a boolean or None value. In case it is unknown, it should be `None`.

So I think this is a correct fix for the current stable version, but incorrect behavior for the current dev branch.

Reply:
haha! In that case it was already working correctly.
In fact it is unknown because the sensor is disabled in Hue. Even Hue reports it as unknown due to the None value.

Reply:
> In fact it is unknown because the sensor is disabled in Hue. Even Hue reports it as unknown due to the None value.

Hmm... so shouldn't that be marked as unavailable on the Home Assistant end in that case?

Reply:
Good suggestion but unavailable is used for connection state...

Reply:
So unavailable on entity/device level, when connecting to hubs, is generally a combined state.

If the Hub is unavailable/cannot be connected to, of course, the device and its entities won't be available either. However, if the device is not available on the hub, the device and entities can be marked unavailable as well, while other devices of the hub remain available.

Reply:
@frenck I've adjusted the logic to only return False if the sensor is disabled (hue specific).
The other code stays the same so a None value will stay None

Reply:
> So unavailable on entity/device level, when connecting to hubs, is generally a combined state.
> 
> If the Hub is unavailable/cannot be connected to, of course, the device and its entities won't be available either. However, if the device is not available on the hub, the device and entities can be marked unavailable as well, while other devices of the hub remain available.

In Hue we even know at device level if it's available due to the zigbee status.
But the sensor is not unavailable, it's disabled (which is a bit Hue specific I guess).

Anyways I think I managed to fix it

Reply:
if I may add please, from a user perspective a binary motion sensor should either be Clear, of Detected. Even when switched off, it would be Clear (as it is currently after a single time turning on/off again.)

What's the added value of returning unknown if the user has switched off the sensor themselves?

Reply:
> , from a user perspective a binary motion sensor should either be Clear, or Detected. (or unavailable when in trouble..)

And as of Home Assistant 2022.2, Unknown when we don't know.

> What's the added value of returning unknown if the user has switched off the sensor themselves?

That is a different discussion, which was already in progress above.

Reply:
> In Hue we even know at device level if it's available due to the zigbee status.
> But the sensor is not unavailable, it's disabled (which is a bit Hue specific I guess).

Isn't that the same? If it is disabled in Hue, it won't be available for use in Home Assistant. IMHO, that should be rendered as unavailable.

Reply:
Not exactly as the other attributes and entities of the device still work. Unavailable is when the device itself is not reachable. At least that's how we implemented it in Hue ;-)

I've adjusted the code to specific check that enabled state, might be enough for now ?

Reply:
I don't agree on that as per above. Sorry.

Reply:
In that case I think we should leave it as-is. Attaching the available state to the enabled property also doesn't feel right as users might go looking for connectivity issues, not remembering they disabled the sensor

Reply:
> And as of Home Assistant 2022.2, Unknown when we don't know.

except that we do know, because their associated switch is turned-off, so it must be Clear (can never be 'Detected')

as said, that feels regression.... A frontend with all unknowns, even though a sensors parent switch is in a known state. And toggling it will fix it.

This has been working fine for years, and now an academic truth (unknown because no positive other state can be determined), while in practice every end user will know if their switch is turned on or off.

Reply:
I really don't agree with that perspective actually. There is no difference here compared to e.g., unplugging a bulb. Communication with the hub is possible, true, but communication with the end device has been cut off by the hub because it has been disabled there. The device is therefore not available.